e2e/release/src/version-plans-filtered-release.test.ts

```diff
@@ -0,0 +1,171 @@
+import { existsSync } from 'fs';
+import { ensureDir, writeFile } from 'fs-extra';
+import { join } from 'path';
+import {
+  cleanupProject,
+  newProject,
+  runCLI,
+  runCommandAsync,
+  tmpProjPath,
+  uniq,
+  updateJson,
+} from '@nx/e2e-utils';
+
+describe('nx release with version plans and project filter', () => {
+  let pkg1: string;
+  let pkg2: string;
+  let pkg3: string;
+
+  const exists = (filePath: string) => existsSync(filePath);
+
+  beforeAll(async () => {
+    newProject({
+      packages: ['@nx/js'],
+      preset: 'ts',
+    });
+
+    pkg1 = uniq('my-pkg-1');
+    runCLI(
+      `generate @nx/js:library ${pkg1} --publishable --importPath=${pkg1}`
+    );
+
+    pkg2 = uniq('my-pkg-2');
+    runCLI(
+      `generate @nx/js:library ${pkg2} --publishable --importPath=${pkg2}`
+    );
+
+    pkg3 = uniq('my-pkg-3');
+    runCLI(
+      `generate @nx/js:library ${pkg3} --publishable --importPath=${pkg3}`
+    );
+
+    // Enable independent versioning with version plans
+    updateJson('nx.json', (nxJson) => {
+      nxJson.release = {
+        ...nxJson.release,
+        projectsRelationship: 'independent',
+        versionPlans: true,
+        version: {
+          conventionalCommits: false,
+          generatorOptions: {
+            specifierSource: 'version-plans',
+          },
+        },
+        changelog: {
+          workspaceChangelog: false,
+          projectChangelogs: {
+            createRelease: false,
+          },
+        },
+      };
+      return nxJson;
+    });
+
+    await runCommandAsync(`git add .`);
+    await runCommandAsync(`git commit -m "chore: initial setup"`);
+  }, 120000);
+
+  afterAll(() => cleanupProject());
+
+  it('should only delete version plans that exclusively apply to filtered projects', async () => {
+    const versionPlansDir = tmpProjPath('.nx/version-plans');
+    await ensureDir(versionPlansDir);
+
+    // Create version plan for pkg1 only
+    await writeFile(
+      join(versionPlansDir, 'bump-pkg1.md'),
+      `---
+${pkg1}: minor
+---
+
+Update package 1 with a minor bump
+`
+    );
+
+    // Create version plan for pkg2 only
+    await writeFile(
+      join(versionPlansDir, 'bump-pkg2.md'),
+      `---
+${pkg2}: patch
+---
+
+Update package 2 with a patch bump
+`
+    );
+
+    // Create version plan for pkg3 only
+    await writeFile(
+      join(versionPlansDir, 'bump-pkg3.md'),
+      `---
+${pkg3}: major
+---
+
+Update packages 3 with major
+`
+    );
+
+    await runCommandAsync(`git add ${versionPlansDir}`);
+    await runCommandAsync(`git commit -m "chore: add version plans"`);
+
+    // Verify all version plans exist before release
+    expect(exists(join(versionPlansDir, 'bump-pkg1.md'))).toBeTruthy();
+    expect(exists(join(versionPlansDir, 'bump-pkg2.md'))).toBeTruthy();
+    expect(exists(join(versionPlansDir, 'bump-pkg3.md'))).toBeTruthy();
```

Comment:
Nit: Could maybe save these paths to a var to just be 100% sure that they are identical before and after